### ISSUE TITLE
make gdx Color Serializable

### ISSUE THREAD

**Issue by judos** [libgdx/libgdx#6641]
#### Issue details
It would help for storing save files to make the class [graphics.Color](https://github.com/libgdx/libgdx/blob/master/gdx/src/com/badlogic/gdx/graphics/Color.java) implement the Serializable interface.

I would suppose there is no downside in having this. While I'm still new to libGdx maybe someone else sees more classes which could use this.

#### Version of libGDX and/or relevant dependencies
1.9.14
Quick check in the master branch shows that the newest version is affected as well.

#### Please select the affected platforms
Doens't really matter which platform I think :)

Thanks for this awesome library!

**Comment by tommyettinger** [comment]
This is a bad idea. See this Stack Overflow answer for context, https://stackoverflow.com/questions/4548816/when-should-we-implement-serializable-interface/53889695#53889695 , and this summary of the relevant section in Effective Java, https://qtips.github.io/2017/08/effective-java-chapter-11 . The java.io.Serializable interface is used very little in libGDX, with all usage in the `com.badlogic.gdx.math` and `com.badlogic.gdx.math.collision` packages. That means any code that uses other libGDX packages, like the `com.badlogic.gdx.utils` package with its ubiquitous Array class, won't be Serializable. Using the provided Json class for serialization is a good option if you want human-readable output, or you can use [Kryo](https://github.com/EsotericSoftware/kryo) for a better approach for binary output. You can also read and write plain RGBA8888 ints that each Color can produce, which may be the best option for size reasons. I'm closing this because Serializable is a bigger problem than a feature, according to experts on Java and the people in charge of the language's future, on top of the significant number of severe disclosed vulnerabilities relating specifically to Java's built-in serialization. These vulnerabilities can also be related to games, because they are a common way network traffic is interfered with or save-files are tampered with.

tl;dr don't use Java's built-in serialization, use libGDX's Json or the related Kryo library instead.

**Comment by judos** [comment]
Thank you for the indepth explanation and further links provided!
